dotbot/plugins/conditional.py

```diff
@@ -0,0 +1,43 @@
+import dotbot
+from dotbot.dispatcher import Dispatcher
+from dotbot.tester import Tester
+
+class Conditional(dotbot.Plugin):
+
+    '''
+    Conditionally execute nested commands based on the result of configured test(s)
+    '''
+
+    _directive = "conditional"
+
+    def can_handle(self, directive):
+        return directive == self._directive
+
+    def handle(self, directive, data):
+        if directive != self._directive:
+            raise ValueError("Conditional cannot handle directive %s" % directive)
+        return self._process_conditional(data)
+
+    def _process_conditional(self, data):
+        success = True
+        tests = data.get("if")
+        test_result = Tester(self._context).evaluate(tests)
+
+        tasks = data.get("then") if test_result else data.get("else")
+        self._log.info("executing sub-commands")
+        # TODO prepend/extract defaults if scope_defaults is False
+        if tasks is not None:
+            return self._execute_tasks(tasks)
+        else:
+            return True
+
+    def _execute_tasks(self, data):
+        # TODO improve handling of defaults either by reusing context/dispatcher -OR- prepend defaults & extract at end
```

Comment:
Note this currently has the same issue as the similar third-party plugins (scoping defaults to the child tasks). I think the best approach would be to update dispatcher constructor to allow passing in (and therefore reusing) a context.

Other options considered:
- Reusing the parent dispatcher by exposing it through context
- Modify the task list by prepending a default directive. After completion, get the child context defaults and call set_defaults on parent context.

In either case, I also think it's worth adding a configuration option to this plugin to control whether defaults are scoped to the child tasks or not.